### PR TITLE
Automated publishing, Makefile cleanup, Black, improved release instructions

### PR DESCRIPTION
**Pull request by Stormheg**
Attempts to declutter the project and ease maintenance.

- Remove a lot of targets from the `Makefile` which I believe are rarely used. Less is better.
- ~~Remove the Github Pages deployment, this does not appear to be used~~ _actually used to build the the `latest` version on Read the Docs_
- Add [Trusted Publishing](https://docs.pypi.org/trusted-publishers/) and trigger PyPI releases from a GitHub release. Meaning anyone with maintainer access to this repository can make releases. No need for PyPI tokens.
- Check linting of Javascript/CSS files in CI
- Add Black - there is not much Python code in this library but it is nice to have consistency.
- Remove [`zest.releaser`](https://zestreleaser.readthedocs.io/). (which provides the `prerelease` and `postrelease` commands mentioned in the original documentation). We don't use tools like this much in other projects and I think it is best to just remove it (for consistency and to avoid confusion)

### CODE REVIEW

**Comment by Stormheg** [comment]
I've just set up the Trusted Publisher in the PyPI interface.

**Comment by Stormheg** [comment]
I'm not sure why Vercel is failing. I don't have access and cannot view the logs.

**Comment by Stormheg** [comment]
I stumbled on https://github.com/wagtail/sphinx-wagtail-theme/issues/226#issuecomment-1462422734 which mentions that the `latest` version on read the docs is actually built from the `gh-pages`. Removing the GitHub Pages build from the CI will break that integration

Looks like I'll have to consider if there is an alternate way of building the latest version of the theme, or bring back the Pages build.

**Comment by laymonage** [comment]
@Stormheg I fixed the readthedocs config to also build the frontend assets in #284, so I think that might be irrelevant now. I suppose the reason why we had the Vercel setup was also because we didn't have the setup to build the frontend assets for PR builds on readthedocs. Now that it's been fixed, we can probably remove both of them.

**Comment by lb-** [comment]
Thanks both for making this better.

**Comment by Stormheg** [comment]
@laymonage I suppose you are right and we don't need Veercel anymore. I've removed the integration.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+name: Publish to PyPI
+on:
+  release:
+    # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release
+    types: [released]
+jobs:
+  test:
+    uses: ./.github/workflows/co.yml
```
Comment (laymonage) [inline .github/workflows/publish.yml:8]:
```suggestion
    uses: ./.github/workflows/ci.yml
```

**Review by laymonage — COMMENTED** [review]
Thanks! I'll rebase this to see if everything still works as expected, then do a full review.

**Review by laymonage — APPROVED** [review]
Looks good to me, thanks @Stormheg!

**Pull request merged**